api/api/src/main/scala/uk/ac/wellcome/platform/api/models/Aggregation.scala

```diff
@@ -3,39 +3,61 @@ package uk.ac.wellcome.platform.api.models
 import scala.util.Try
 import io.circe.Decoder
 import java.time.{Instant, LocalDateTime, ZoneOffset}
-import grizzled.slf4j.Logging
-import uk.ac.wellcome.models.work.internal.{Period, WorkType}
-import uk.ac.wellcome.json.JsonUtil._
 
-trait OntologyType {
-  val ontologyType: String
+import grizzled.slf4j.Logging
+import uk.ac.wellcome.models.work.internal.{
+  AbstractConcept,
+  Displayable,
+  Genre,
+  Period,
+  WorkType
 }
-object OntologyType {}
+import uk.ac.wellcome.json.JsonUtil._
 
-// TODO: Return a `ParseError`s where applicable
-case class Genre(label: String)
+case class AggregatedGenre(label: String)
 
-case class Aggregations(workType: Option[Aggregation[WorkType]] = None,
-                        genre: Option[Aggregation[Genre]] = None,
-                        productionDates: Option[Aggregation[Period]] = None)
+case class Aggregations(
+  workType: Option[Aggregation[WorkType]] = None,
+  genres: Option[Aggregation[Genre[Displayable[AbstractConcept]]]] = None,
+  productionDates: Option[Aggregation[Period]] = None)
 
 object Aggregations extends Logging {
 
   def apply(jsonString: String): Option[Aggregations] =
     fromJson[EsAggregations](jsonString)
       .collect {
-        case EsAggregations(workType, genre, date)
-            if List(workType, genre, date).flatten.nonEmpty =>
+        case EsAggregations(workType, aggregatedGenres, date)
+            if List(workType, aggregatedGenres, date).flatten.nonEmpty => {
+
+          // We have to do this conversion here as we only get a label back from the Elastic response
+          // as it creating a composite aggregation, which doesn't really have the knowledge of a full
+          // Genre object
+          val genres = convertAggregationData(
+            getAggregation[AggregatedGenre](aggregatedGenres),
+            (aggregateGenre: AggregatedGenre) =>
+              Genre[Displayable[AbstractConcept]](aggregateGenre.label, List()))
+
           Some(
             Aggregations(
               workType = getAggregation[WorkType](workType),
-              genre = getAggregation[Genre](genre),
+              genres = genres,
               productionDates = getAggregation[Period](date)
             )
           )
+        }
       }
       .getOrElse { None }
 
+  def convertAggregationData[T, V](maybeAgg: Option[Aggregation[T]],
+                                   c: T => V): Option[Aggregation[V]] =
```

Comment:
I reckon `convert` or something else less vague might be a better var name for this?

Reply:
✨